File: frontend/src/dtos/RelatorioDto.ts
export interface RelatorioPessoaDto {
    pessoaId: number;
    pessoaNome: string;
    totalReceitas: number;
    totalDespesas: number;
    saldo: number;
}

export interface RelatorioCategoriaDto {
    categoriaId: number;
    categoriaDescricao: string;
    totalReceitas: number;
    totalDespesas: number;
    saldo: number;
}

export interface RelatorioGeralDto<T> {
    itens: T[];
    totalReceitas: number;
    totalDespesas: number;
    saldo: number;
}

File: ControleGastos/ControleGastos/Application/DTOs/RelatorioDto.cs
namespace ControleGastos.Application.DTOs;

public class RelatorioPessoaDto
{
    public int PessoaId { get; set; }
    public string PessoaNome { get; set; } = string.Empty;
    public decimal TotalReceitas { get; set; }
    public decimal TotalDespesas { get; set; }
    public decimal Saldo { get; set; }
}

public class RelatorioCategoriaDto
{
    public int CategoriaId { get; set; }
    public string CategoriaDescricao { get; set; } = string.Empty;
    public decimal TotalReceitas { get; set; }
    public decimal TotalDespesas { get; set; }
    public decimal Saldo { get; set; }
}

public class RelatorioGeralDto<T>
{
    public List<T> Itens { get; set; } = new List<T>();
    public decimal TotalReceitas { get; set; }
    public decimal TotalDespesas { get; set; }
    public decimal Saldo { get; set; }
}
